compiler/rustc_resolve/src/lib.rs

```diff
@@ -1521,6 +1533,101 @@ impl<'a> Resolver<'a> {
         });
     }
 
+    /// Compute access levels for exports and intermediate use statements
+    fn resolve_access_level(&mut self, krate: &Crate) {
+        self.set_access_level(CRATE_NODE_ID, Some(AccessLevel::Public));
+
+        self.set_exports_access_level(CRATE_DEF_ID);
+        let mut privacy_visitor =
+            access_levels::PrivacyVisitor::new(self, Some(AccessLevel::Public));
+
+        // We need a double pass to correctly resolve exports
+        // as an item can be defined in a private module
+        // and exported later on
+        //
+        // mod foo {
+        //     pub enum Hungry;
+        // }
+        //
+        // pub mod bar {
+        //     pub use super::foo::*;
+        // }
```

Comment:
I'm pretty sure it should be a loop and not a double pass because cases like
```rust
mod privmod1 {
    pub struct Orig;
}

mod privmod2 {
    pub use foo::Orig as Reexport1;
}

pub pubmod {
    pub use privmod2::Reexport1 as Reexport2;
}
```
(with an arbitrary number of private modules) are possible.